File: in_Java/0374 Guess Number Higher or Lower.java
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return       -1 if num is lower than the guess number
 *                1 if num is higher than the guess number
 *               otherwise return 0
 * int guess(int num);
 */

public class Solution extends GuessGame {
    public int guessNumber(int n) {
        int i = 1; 
        int j = n;
        while(i <= j){
            int m = i + (j-i)/2;
            if(guess(m) == 0) return m;
            else if(guess(m) == 1) i = m + 1;
            else j = m-1;
        }
        return -1;
    }
}

File: in_Python/0374 Guess Number Higher or Lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num):

class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        i = 1
        j = n
        while i <= j:
            m = i + (j-i) / 2
            if guess(m) == 0:
                return m
            elif guess(m) == 1:
                i = m + 1
            else:
                j = m - 1
        return -1
        